### DISCUSSION TITLE
how to use custom config to set token

### DISCUSSION THREAD

**Discussion by liompei** [coil-kt/coil#1340] [General]
like GlideModule ,
i need use token with in all image request

**Comment by cemore2048** [comment]
are you talking about an authentication token? You can add a custom interceptor the all the coil requests. There's a pretty good example of how to do here:

https://coil-kt.github.io/coil/image_pipeline/#interceptors

**Comment by liompei** [comment]
So I need to add this code manually every time I want to use it, instead of just writing it once in the glidemodule like glide

**Comment by cemore2048** [comment]
No, you write it once and then add it to your image loader

```kotlin
val imageLoader = ImageLoader.Builder(context)
    .components {
        add(YourCustomInterceptorHere())
    }
    .build()
```
